src/typer/tree.ml

```diff
@@ -50,7 +50,7 @@ and type_bind =
       (* exposed env *)
       env : Env.t;
       loc : Location.t;
-      type_ : Type.t;
+      field : Type.field;
```

Comment:
This change is unrelated and I added it by accident, but essentially it makes so that TT_bind has access to it's own forall, which is pretty good.